llvm/test/CodeGen/X86/setcc-lowering.ll

```diff
@@ -9,10 +9,11 @@
 define <8 x i16> @pr25080(<8 x i32> %a) {
 ; AVX-LABEL: pr25080:
 ; AVX:       # %bb.0: # %entry
+; AVX-NEXT:    vandps {{\.?LCPI[0-9]+_[0-9]+}}(%rip), %ymm0, %ymm0
+; AVX-NEXT:    vcvtdq2ps %ymm0, %ymm0
+; AVX-NEXT:    vxorps %xmm1, %xmm1, %xmm1
+; AVX-NEXT:    vcmpeqps %ymm1, %ymm0, %ymm0
```

Comment:
This is the one remaining regression pattern. Basically if demandedelts is only contained the xmm, this will be a regression.

I was thinking maybe move the entire transform to `SimplifyDemandedBits` and only do if `DemandedElts` contains `ymm`, but the transform is pretty x86_64 exclusive so wasn't sure if it fit.

Any ideas @RKSimon

Reply:
I'll take a look

Reply:
I'm not sure how we can easily avoid the conversion, but I've pushed a couple of changes that should help - please can you rebase? I think the vcvtdq2ps might still be a ymm but we should be getting better.

Reply:
Rebased and at least it stays in `xmm` now. I think what we could do, is I could make the transform reversible. Then we could have rules for int->float and float->int.

Reply:
@RKSimon , what do you think?